src/views/common/Animation/AnimationControls/usePlaybackBarSelection.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useMemo, useRef } from 'react'
```

Comment:
Implements the ability to drag-select parts of the scrub bar in order to create a sub-window, basically zooming in on a particular section of the animation.